test/rules/assertions/requireParam.js

```diff
@@ -442,10 +516,18 @@ export default {
         {
           message: 'Missing JSDoc @param "foo" declaration.',
         },
+        {
+          message: 'Missing JSDoc @param "root.bar" declaration.',
+        },
+        {
+          message: 'Missing JSDoc @param "root.baz" declaration.',
+        },
       ],
       output: `
       /**
        * @param foo
+       * @param root.bar
+       * @param root.baz
        */
       function quux (foo, {bar, baz}) {
       }
```

Comment:
I think we need three things here:

1. Tests for `function quux ({foo}, {bar, baz}) {` (but with no existing `@param foo` tag) to ensure we are not adding `root` for both parameters.
1. Option to avoid a fixer for missing roots (some might not want variable names injected)
1. Configuration to change the name from "root", e.g., an array of preferred (`["root", "cfg", "config"]`) and when these are exhausted, auto-appending numbers (e.g., `root2`)

Reply:
The last is less critical (esp. with no. 2) implemented.

Reply:
> 1. Tests for `function quux ({foo}, {bar, baz}) {` (but with no existing `@param foo` tag) to ensure we are not adding `root` for both parameters.

Oh, great catch! In this instance, would the names be `root.foo, root1.bar, root1.baz`?

> 2. Option to avoid a fixer for missing roots (some might not want variable names injected)

Would this option disable the fixer for all missing parameters, or just those that are destructured?

> 3. Configuration to change the name from "root", e.g., an array of preferred names (`["root", "cfg", "config"]`) and when these are exhausted, auto-appending numbers (e.g., `root2`)

So, in the example from 1, the names would be `root.foo, cfg.bar, cfg.baz`? I'm not sure I understand how that would be useful. I can see customizing what `root` is (for example, Kotlin uses `p`), so I definitely support customizing, but I'm not sure what providing more than one option accomplishes.

Just a bit more info on Kotlin: it uses `p{int}` as parameter names. So, the first example would be `p0.foo, p1.bar, p1.baz`. I'll defer to your judgement as to whether we should append a `0` for the first parameter.

Reply:
> In this instance, would the names be `root.foo, root1.bar, root1.baz`?

Yeah, good question, I guess so. One could argue for immediately jumping to `root2` or changing `root` to `root1` or as you mention for Kotlin, using "0". As long as the fixer is disable-able, I'd personally be fine with whatever choice (though maybe having all as numbers is good when they are being used to avoid the impression that the numbered ones are to be understood as a subset of the unnumbered one somehow).

> Would this option disable the fixer for all missing parameters, or just those that are destructured?

If we're auto-adding `"root"` as a name for non-destructured parameters, then yes, it should disable that behavior too. But I thought that was a new introduction only, and in such a case, it is fine to apply just for destructured.

> So, in the example from 1, the names would be `root.foo, cfg.bar, cfg.baz`?

Right, that was the idea. Maybe a better example would be `["config", "cfg", "options"]` where the names are generic enough but related (e.g., for my codebases which tend to use object options). The advantage would be just that it can avoid adding numbers.